File: examples/node/minimal.js
'use strict';

const path = require('path');
const bazi = require(path.join(__dirname, '../../bindings/js'));

const result = bazi.calculate({ date: '1990-05-15', hour: 12 });
console.log('ganzhi:', result.bazi_ganzhi);

const dayun = bazi.dayun({ date: '1990-05-15', gender: 'male', count: 3 });
console.log('dayun count:', dayun.dayun_list.length);

const ln = bazi.liunian({ year: 2024, dayStem: result.pillars.day.stem });
console.log('liunian:', ln.ganzhi, ln.shishen || '');

File: examples/python/minimal.py
#!/usr/bin/env python3
"""Minimal Python example using the installed package."""
from datetime import date

from mystilink_bazi import compute_bazi, compute_dayun, compute_liunian

birth = date(1990, 5, 15)
result = compute_bazi(birth, hour_interval=12, minute=0)
print("ganzhi:", result["bazi_ganzhi"])

dayun = compute_dayun(birth, "male", count=3)
print("dayun count:", len(dayun["dayun_list"]))

liunian = compute_liunian(2024, day_stem=result["pillars"]["day"]["stem"])
print("liunian:", liunian["ganzhi"], liunian.get("shishen"))
